msgpack/src/main/scala/fs2/data/msgpack/low/internal/FormatParsers.scala

```diff
@@ -0,0 +1,131 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data
+package msgpack
+package low
+package internal
+
+import fs2.data.msgpack.low.MsgpackItem
+import fs2.data.msgpack.low.internal.Helpers._
+import scodec.bits._
+
+private[internal] object FormatParsers {
+  def parseSimpleType[F[_]](lift: ByteVector => MsgpackItem)(length: Int, ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    requireBytes(length, ctx) map { res =>
+      res.accumulate(bytes => lift(bytes))
+    }
+  }
+
+  def parseUnsignedInt[F[_]](length: Int, ctx: ParserContext[F])(implicit F: RaiseThrowable[F]) =
+    parseSimpleType(MsgpackItem.UnsignedInt)(length, ctx)
+
+  def parseArray[F[_]](length: Int, ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    requireBytes(length, ctx) map { res =>
+      res.accumulate(v => MsgpackItem.Array(v.toInt(false, ByteOrdering.BigEndian)))
+    }
+  }
+
+  def parseMap[F[_]](length: Int, ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    requireBytes(length, ctx) map { res =>
+      res.accumulate(v => MsgpackItem.Map(v.toInt(false, ByteOrdering.BigEndian)))
+    }
+  }
+
+  def parseTimestamp[F[_]](length: Int, ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    length match {
+      case 4 =>
+        requireBytes(4, ctx) map { res =>
+          res.accumulate(v => MsgpackItem.Timestamp32(v))
+        }
+      case 8 =>
+        requireBytes(8, ctx) map {
+          case res => {
+            val result = res.result
+            val seconds = result & hex"00000003ffffffff"
+            val nanosec = result >> 34
+
+            res.toContext.prepend(MsgpackItem.Timestamp64(nanosec.drop(4), seconds.drop(3)))
+          }
+        }
+      case 12 =>
+        for {
+          res <- requireBytes(4, ctx)
+          nanosec = res.result
+          res <- requireBytes(8, res.toContext)
+          seconds = res.result
+        } yield res.toContext.prepend(MsgpackItem.Timestamp96(nanosec, seconds))
+      case _ => Pull.raiseError(new MsgpackParsingException(s"Invalid timestamp length: ${length}"))
+    }
+  }
+
+  def parseFixExt[F[_]](length: Int, ctx: ParserContext[F])(implicit
+      F: RaiseThrowable[F]): Pull[F, MsgpackItem, ParserContext[F]] = {
+    requireOneByte(ctx) flatMap {
+      res => {
```

Comment:
```suggestion
    requireOneByte(ctx).flatMap { res => 
```

No need for an extra block after `res =>`. Also, I'd prefer if we could avoid infix `flapMap` (and alike) for a) consistency and b) because Scala 3 only wants infix usage of methods defined explicitly as infix. Not keen on having to fix all occurrences with an upcoming Scala update…this obviously applies to all occurrences, just picked this line as an example.

Reply:
Makes sense! Fixed in afa91e1.